test/private/testMatlab2tikz.m

```diff
@@ -0,0 +1,151 @@
+%   passes the cell array of options to MATLAB2TIKZ. Default: {}
+%
+% TESTMATLAB2TIKZ('figureVisible', LOGICAL, ...)
+%   plots the figure visibly during the test process. Default: false
+%
+% TESTMATLAB2TIKZ('testsuite', FUNCTION_HANDLE, ...)
+%   Determines which test suite is to be run. Default: @ACID
+%   A test suite is a function that takes a single integer argument, which:
+%     when 0: returns a cell array containing the N function handles to the tests
+%     when >=1 and <=N: runs the appropriate test function
+%     when >N: throws an error
+%
+% See also matlab2tikz, ACID
+
+% Copyright (c) 2008--2014, Nico Schlömer <nico.schloemer@gmail.com>
+% All rights reserved.
+%
+% Redistribution and use in source and binary forms, with or without
+% modification, are permitted provided that the following conditions are met:
+%
+%    * Redistributions of source code must retain the above copyright
+%      notice, this list of conditions and the following disclaimer.
+%    * Redistributions in binary form must reproduce the above copyright
+%      notice, this list of conditions and the following disclaimer in
+%      the documentation and/or other materials provided with the distribution
+%
+% THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+% AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+% IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+% ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+% LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+% CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+% SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+% INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+% CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+% ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+% POSSIBILITY OF SUCH DAMAGE.
+%
+% =========================================================================
+
+  % In which environment are we?
+  env = getEnvironment();
+
+  % -----------------------------------------------------------------------
+  ipp = m2tInputParser;
+
+  ipp = ipp.addOptional(ipp, 'testFunctionIndices', [], @isfloat);
+  ipp = ipp.addParamValue(ipp, 'extraOptions', {}, @iscell);
+  ipp = ipp.addParamValue(ipp, 'figureVisible', false, @islogical);
+  ipp = ipp.addParamValue(ipp, 'actionsToExecute', @(varargin) varargin{1}, @isFunction);
+  ipp = ipp.addParamValue(ipp, 'testsuite', @ACID, @isFunction );
+
+  ipp = ipp.parse(ipp, varargin{:});
+
+  ipp = sanitizeInputs(ipp);
+
+  % -----------------------------------------------------------------------
+  stdout = 1;
+  if strcmp(env, 'Octave') && ~ipp.Results.figureVisible
+      % Use the gnuplot backend to work around an fltk bug, see
+      % <http://savannah.gnu.org/bugs/?43429>.
+      graphics_toolkit gnuplot
```

Comment:
I don't have a Windows environment with Octave installed at my disposal right not, but for me the `strcmp` raises an orange flag: it might be safer to use `strcmpi`.